### PR TITLE
Update to CE 1.10.1

### PR DESCRIPTION
**Pull request by tmewett**
Thanks sluedecke and apollovy!

For #1

### CODE REVIEW

File: iBrogueCE_iPad/BrogueCode/IO.c [hunk 1]
```diff
@@ -922,7 +923,7 @@ void refreshScreen() {
     for (int i = 0; i < COLS; i++) {
         for (int j = 0; j < ROWS; j++) {
             cellDisplayBuffer *curr = &displayBuffer[i][j];
-            plotChar(curr->character, i, j,
+            plotChar(glyphToUnicode(curr->character), i, j,
```
Comment (tmewett) [inline iBrogueCE_iPad/BrogueCode/IO.c:926]: Can this go in the iPad side - to minimise CE diff?

File: iBrogueCE_iPad/BrogueCode/IO.c [hunk 2]
```diff
@@ -2805,7 +2806,7 @@ boolean getInputTextString(char *inputText,
     maxLength = min(maxLength, COLS - x);
 
 
-    strcpy(inputText, defaultEntry);
+//    strcpy(inputText, defaultEntry);
```
Comment (tmewett) [inline iBrogueCE_iPad/BrogueCode/IO.c:2809]: What's this for?

File: iBrogueCE_iPad/BrogueCode/MainMenu.c [hunk 3]
```diff
@@ -501,6 +500,7 @@ boolean dialogChooseFile(char *path, const char *suffix, const char *prompt) {
     qsort(files, count, sizeof(fileEntry), &fileEntryCompareDates);
 
     currentPageStart = 0;
+    brogueButton buttons[min(count - currentPageStart, FILES_ON_PAGE_MAX)];
```
Comment (tmewett) [inline iBrogueCE_iPad/BrogueCode/MainMenu.c:503]: Why this change?

File: iBrogueCE_iPad/BrogueCode/Time.mm [hunk 4]
```diff
@@ -1037,9 +1037,9 @@ void activateMachine(short machineNumber) {
                 && cellHasTMFlag(x, y, TM_IS_WIRED)) {
 
                 pmap[x][y].flags |= IS_POWERED;
-                for (layer = 0; layer < NUMBER_TERRAIN_LAYERS; layer++) {
+                for (layer = DUNGEON; layer < NUMBER_TERRAIN_LAYERS; layer = dungeonLayers(layer + 1)) {
```
Comment (tmewett) [inline iBrogueCE_iPad/BrogueCode/Time.mm:1040]: I am unsure why these are needed, they never were before; TODO for future

File: iBrogueCE_iPad/BrogueCode/platformdependent.c [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+//        strcpy(returnList[i].date,          scoreBuffer[i].dateText);
+//        strcpy(returnList[i].description,   scoreBuffer[i].description);
+//    }
+//
+//    return mostRecentLineNumber;
+//}
+
+//boolean saveHighScore(rogueHighScoresEntry theEntry) {
+//    short i, lowestScoreIndex = -1;
+//    long lowestScore = -1;
```
Comment (tmewett) [inline iBrogueCE_iPad/BrogueCode/platformdependent.c:435]: Could we delete instead of commenting? I'm guessing there's a symbol clash - if there isn't, we can leave them in to minimise diffs & future updating work

File: iBrogueCE_iPad/PlatformCode/RogueScene.swift [hunk 6]
```diff
@@ -162,20 +162,20 @@ fileprivate extension RogueScene {
...
-                    self = .omega
+//                case "\(UnicodeScalar(UInt32(FOLIAGE_CHAR))!)":
+//                    self = .foliage
+//                case "\(UnicodeScalar(UInt32(SCROLL_CHAR))!)":
+//                    self = .scroll
+//                case "\(UnicodeScalar(UInt32(CHARM_CHAR))!)":
+//                    self = .charm
+//                case "\(UnicodeScalar(UInt32(RING_CHAR))!)":
+//                    self = .ring
+//                case "\(UnicodeScalar(UInt32(AMULET_CHAR))!)":
```
Comment (tmewett) [inline iBrogueCE_iPad/PlatformCode/RogueScene.swift:173]: Can this be deleted?

**Review by tmewett — COMMENTED** [review]
Thank you @apollovy, I have just a couple of questions, mostly optimising for reducing diffs/divergence between iPad and upstream CE.

Also if you have the chance, would be good to do couple of quick seed comparisons between PC and iPad and check items and terrain look the same.

File: iBrogueCE_iPad/BrogueCode/IO.c (hunk 1 above)
Comment (apollovy) [reply iBrogueCE_iPad/BrogueCode/IO.c]: Done

File: iBrogueCE_iPad/BrogueCode/Time.mm (hunk 4 above)
Comment (apollovy) [reply iBrogueCE_iPad/BrogueCode/Time.mm:1040]: This one was crapped initially: someone forgot to increment the `layer` variable and everything was hanging in a 100% CPU eating state inside this cycle. I've changed this to simplier version, and the found initial one and applied that now. And it works.

**Comment by sethhoward** [comment]
Let’s take the weekend to play test while I get my account with Apple back in order. When we’re happy with the results give me a thumbs up and I’ll package for release. 

Great work!

**Comment by SeymourJ** [comment]
The “dense foliage”  and “bloodwort stalk” characters don’t draw correctly in the version I downloaded and built yesterday.  They both use the same glyph, but in different colors, and both draw as a weird V like shape.

**Comment by SeymourJ** [comment]
The welcome text “Hello adventurer…” at the start of a game requires two taps of the —more— button to display fully in this build.  In seth’s 1.75 build, it is simply displayed at the top when you start a new game.

also, the “Press <?> for help at any time prompt is not useful on iPad where the vast majority of folks won’t have a keyboard attached, and help is accessed by tapping “Menu” and then the “Help” item.   You could remove the prompt, or change it to “Tap on Menu at the bottom and then tap Help for help at any time.”

**Comment by SeymourJ** [comment]
Upon death, this build prompts with “Press ‘i’ to view your inventory” which, again isn’t useful on a device without a hardware keyboard.  Seth’s builds removed this prompt.  

The “show inventory” button that appears on death in this build,  is disabled by the “—more—“ prompt that appears at the same time.   In Seth’s 1.75 build,  you could tap “show inventory” without tapping “—more—“ first.   The button probably shouldn’t be displayed when you can’t actually tap it.

You have to tap “—more—“ twice to proceed on the “you are dead” page in this build.   Only once in Seth’s 1.75 build.

Sorry if this feels like nitpicking.  I’m thrilled that you’ve gotten us this close to a new build of Brogue for iPad, and I’d like it to be the best it can be.

**Comment by sethhoward** [comment]
These are excellent bug reports. There should be a flag that turns off keyboard prompts and replaces then with appropriate tap text. 

I’d love to see, at some point, the ability to detect a keyboard and toggle that flag!

**Comment by SeymourJ** [comment]
When I call an object something like “known good” it ends up labeled as “u<space><space>known good”

**Comment by SeymourJ** [comment]
Upon reloading a suspended game, the current build leaves a white tap button labeled “esc”. covering the upper left of the screen.  It never goes away.  Also for some reason, it brings up the iPad on-screen keyboard despite no text input being requested.  The on-screen keyboard is easily dismissed.

**Comment by SeymourJ** [comment]
Did I scare everyone off?  It's been about a month since the last commit to Apollovy's branch of Brogue CE for iPad.  Is something amiss?  Are we deadlocked waiting for Seth or Yuri to make the next move or is something else going on?  Sorry to seem impatient, but I'm eager to get this app back in the iOS app store.

**Comment by sethhoward** [comment]
Not at all. I’ve been pretty busy. What I’d like to do is add all of your bugs as ‘issues’ if that hasn’t been done already but haven’t gotten to it. If anyone has a moment and wants to do that..

**Comment by AndrewSav** [comment]
@sethhoward are you planning to merge this PR after the issues are created?

**Comment by sethhoward** [comment]
The issues need to be resolved first. I figure issues will help track what has been fixed and retested.

**Comment by SeymourJ** [comment]
Seth, et al,

I’ve entered the issues I previously reported into "Issues “ on https://github.com/sethhoward/iBrogueCE/issues

I’m kind of new to this so I hope I did it properly.

Seymour

**Comment by SeymourJ** [comment]
Another month with no updates?

**Comment by SeymourJ** [comment]
Added new issue for bug where the Dive into the Depths? prompt repeats infinitely sometimes when you tap "No".

**Comment by SeymourJ** [comment]
Happy New Year all.

**Comment by SeymourJ** [comment]
No changes to Apollov's code branch since late October.   I thought everyone was excited to get this back in the app store, but it seems the project has been idle for more than 90 days.  I'd fix the code myself if I could.  Is there anything else I can do?

**Comment by AndrewSav** [comment]
As I understand it, it just needs someone picking it up and doing the work. It does not seem we have any volonteers currently.

**Comment by tmewett** [comment]
I'm thinking of advertising for a volunteer contributor/maintainer for this port somewhere online. Sorry that progress is slow, but we haven't forgotten about it. Thanks for your help so far with testing and reporting issues

**Comment by sethhoward** [comment]
Last we left off we have a bunch of issues in the tracker. My request was an update to new ver. of Brogue CE and all issues cleared before I’d package it up and release on the App Store which is all I have currently have time for. The offer still stands!

**Comment by SeymourJ** [comment]
What happened to Appolov?  They made such great progress in a couple of weeks last year and then just stopped…

Seymour

**Comment by apollovy** [comment]
Hey all. That's simple: I'm not playing brogue, so the initial motivation was rather low, but there was a challenge to fix major issues when the app was not building at all. When things were more than less working, the motivation went away and I switched to other projects. Sorry, but that's how it works for me 🤓

Kind regards,
Yury Apollov

> 4 февр. 2022 г., в 07:41, SeymourJ ***@***.***> написал(а):
> 
> ﻿
> What happened to Appolov? They made such great progress in a couple of weeks last year and then just stopped…
> 
> Seymour
> 
>

**Comment by SeymourJ** [comment]
The "Press <?> for Help issue on iPad can be resolved by changing line 98 of Rogue.h 

Current:
97  // set to false to disable references to keystrokes (e.g. for a tablet port)
98  #define KEYBOARD_LABELS true

Fix:
97 // set to false to disable references to keystrokes (e.g. for a tablet port)
98 #define KEYBOARD_LABELS false

**Comment by btaylor84** [comment]
I have finally set up an environment to work on this program. Since I'm a newbie Git user (as well as a newbie Swift coder, and Apple Xcode user, etc ... ), I had to fork this to my own repo. That said, I merged this PR, and fixed the foliage and ring issues. If people are interested, my fork is at: https://github.com/btaylor84/ibce. The "working" branch is currently the up-to-date one. I'll continue to work through the issues already identified by https://github.com/SeymourJ and others.

**Comment by SeymourJ** [comment]
Yay Bob!  Thanks for working on Brogue for iPad.

**Comment by SeymourJ** [comment]
Downloaded and built Bob's working branch, tested in both emulator and Pad 8, and can confirm he's resolved the foliage and ring glyph issues.   It also seems to incorporate my fix above for  the "Press <?> for Help" issue.

**Comment by AndrewSav** [comment]
Well done @SeymourJ and @btaylor84, thank you for your effort

**Comment by SeymourJ** [comment]
Wow! Bob is making great progress.   Re-downloaded and built the latest code in his working branch and can confirm that he's fixed the "Quit Menu" issue, and the Double Tap required at the start of game.    It looks like he also fixed the garbled characters when you "Call" an inventory item something.   Yay!

Getting excited.  Very few items left to fix.

**Comment by btaylor84** [comment]
I have fixed all the existing issues in my icbe/working branch. It's been a long time since @sethhoward offered to put this up on the App store, so I'm not sure that's still on the table. If it is, help me submit a pull request, and you can go forward with these revisions.

I am willing to sign up for an Apple Developer account, and work my way through the submission process, but this will be new territory for me, so ... no guarantees. 

@tmewett : I did compare seeds from this build and the Mac version, and they seemed identical.

**Comment by sethhoward** [comment]
Absolutely still stands. When we think we’re code complete let’s make a PR so I can review the code changes and start doing some last minute testing.

**Comment by btaylor84** [comment]
> Edit: which would appear to be this branch. Just give me a thumbs up when we think we’re ready for review. 😁


Ok, I missed this edit. I've been waiting for the approval on the submitted PR. This is ready for review and merge into your code base.

I have been working on another branch that implements keyboard functionality, and I think that's ready for a PR too, but I didn't want to submit another until this was approved.

All in all, this PR puts us back to parity with the former iPad app, and the next PR will add functionality that didn't exist before.

Thanks for taking point on this, @sethhoward , though my offer to pursue a developer account still stands.

**Comment by tmewett** [comment]
Closing this PR now, in favour of the new one
